File: code/python/subarraySum.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        dic = {0:1}
        total = 0
        res = 0
        
        for i in range(len(nums)):
            total += nums[i]
            if (total - k) in dic:
                res += dic[total - k]
            if total in dic:
                dic[total] += 1
            else:
                dic[total] = 1
                
     
        return res

File: code/java/subarraySum.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int res = 0;
        for(int i=0;i<nums.length;i++){
            int j =  i;
            int count = nums[i];
            while(true){
                if(count ==  k)
                    res += 1;
                j = j + 1;
                if(j  == nums.length)
                    break;
                count = count + nums[j];
            }
        }  
        return res;
    }
}
